article/app/liveblog/LiveBlogCurrentPage.scala

```diff
@@ -1,10 +1,71 @@
 package liveblog
 
-import model.liveblog.BodyBlock
+import model.liveblog.{Blocks, BodyBlock}
+
+sealed trait BlockRange { def query: Option[Seq[String]] }
+case object Canonical extends BlockRange {
+  val firstPage = "body:latest:30"
+  val oldestPage = "body:oldest:1"
+  val timeline = "body:key-events"
+  // this only makes sense for liveblogs at the moment, but article use field body not blocks anyway
+  val query = Some(Seq(firstPage, oldestPage, timeline))
+}
+case class PageWithBlock(page: String) extends BlockRange {
+  // just get them all, the caching should prevent too many requests, could use "around"
+  val query = None
+}
+case object ArticleBlocks extends BlockRange {
+  // we currently only use this for emails, we just want all the blocks
+  val query = None
+}
+case class SinceBlockId(lastUpdate: String) extends BlockRange {
+  val around = s"body:around:$lastUpdate:5"
+  // more than 5 could come in (in one go), but unlikely and won't matter as it'll just fetch again soon
+  val query = Some(Seq(around))
+}
 
 object LiveBlogCurrentPage {
 
-  def apply(pageSize: Int, blocks: Seq[BodyBlock], isRequestedBlock: Option[String]): Option[LiveBlogCurrentPage] = {
+  def apply(pageSize: Int, blocks: Blocks, range: BlockRange): Option[LiveBlogCurrentPage] = {
+    range match {
+      case Canonical => firstPage(pageSize, blocks)
+      case PageWithBlock(isRequestedBlock) => findPageWithBlock(pageSize, blocks.body, isRequestedBlock)
+      case SinceBlockId(blockId) => updates(blocks, SinceBlockId(blockId))
+      case ArticleBlocks => None
+    }
+  }
+
+  // filters newer blocks out of the list
+  def updates(blocks: Blocks, sinceBlockId: SinceBlockId) = {
+    val bodyBlocks = blocks.requestedBodyBlocks.get(sinceBlockId.around).toSeq.flatMap { bodyBlocks =>
+      bodyBlocks.takeWhile(_.id != sinceBlockId.lastUpdate)
+    }
+    Some(LiveBlogCurrentPage(FirstPage(bodyBlocks), None))// just pretend to be the first page, it'll be ignored
```

Comment:
ah ok that's just because of the code structure, we have to turn a capi response into a frontend model in order to render it.  Normally this would be all the info needed for the article, liveblog, gallery, front, etc.  However now that  the liveblog model includes knowledge about which page we're on, and only includes a subset of the blocks, the liveblog model doesn't quite work for "updates since block-x" as there's no concept of "which page we're on".
I could have added a model type for LiveBlogUpdatesSince, but I thought for simplicity I'd just call it a FirstPage and embed the blocks in there.  The fact that there's no pagination and various other things doesn't matter to the code that eventually produces the json.